runtime/src/main/java/io/novafoundation/nova/runtime/multiNetwork/connection/ChainConnection.kt

```diff
@@ -109,7 +110,30 @@ class ChainConnection internal constructor(
             .launchIn(this)
     }
 
-    private suspend fun observeCurrentNode() {
+    private fun getNodeUrlFlow(chain: Chain): Flow<NodeWithSaturatedUrl?> {
+        return if (chain.autoBalanceEnabled) {
+            getAutobalancedNodeUrlFlow(chain)
+        } else {
+            val node = chain.nodes.nodes.firstOrNull { it.unformattedUrl == chain.defaultNodeUrl }
+                ?.saturateNodeUrl(connectionSecrets)
+
+            if (node == null) {
+                getAutobalancedNodeUrlFlow(chain)
+            } else {
+                flowOf { node }
+            }
+        }
+    }
```

Comment:
This logic should be in `AutoBalanceStrategyProvider`, though we should rename it to something like `NodeSelectionStrategyProvider`